numpyro/distributions/continuous.py

```diff
@@ -2438,3 +2439,72 @@ def cdf(self, value):
 
     def icdf(self, value):
         return self._ald.icdf(value)
+
+
+class ZeroSumNormal(TransformedDistribution):
+    r"""
+    Zero Sum Normal distribution adapted from PyMC [1] as described in [2,3]. This is a Normal distribution where one or
+    more axes are constrained to sum to zero (the last axis by default).
+
+    Sample code for using ZeroSumNormal in the context of a single axis to zero-contrain::
+
+        def model(category_ind, y): # category_ind is an indexed categorical variable with 20 categories
```

Comment:
This might need `>>>` to be picked up by the doctests. Do you want to call the model so it's execution is validated by doctests?

Reply:
I wasnt planning on it but this looks like a good idea to me I can adjust this

Reply:
added!